sdk/core/Azure.Core.TestFramework/src/ProxyTransport.cs

```diff
@@ -66,6 +66,11 @@ public ProxyTransport(TestProxy proxy, HttpPipelineTransport transport, TestReco
 
         private async Task ProcessAsyncInternalAsync(HttpMessage message, bool async)
         {
+            if (_recording.Mode == RecordedTestMode.Playback && _filter() == EntryRecordModel.DoNotRecord)
+            {
+                throw new InvalidOperationException("Operations that are enclosed in a Do Not Record scope should not be executed in Playback mode." +
```

Comment:
"in a Do Not Record scope" reads a little funny. Maybe just replace "Do Not Record scope" with the actual scope name i.e., `TestRecording.DisableRecordingScope` e.g., "Operations that are enclosed in a TestRecording.DisableRecordingScope should not ...". Or could even mention the methods you'd call in case people are using `var`.